ambari-agent/src/main/python/ambari_agent/CustomServiceOrchestrator.py

```diff
@@ -456,6 +459,43 @@ def runCommand(self, command_header, tmpoutfile, tmperrfile, forced_command_name
       if incremented_commands_for_component:
         self.commands_for_component_in_progress[cluster_id][command['role']] -= 1
 
+      # Conditionally remove the command-*.json file if it exists
+      if os.path.exists(json_path):
+        command_file_retention_policy = self.config.command_file_retention_policy
+
+        if command_file_retention_policy == self.config.COMMAND_FILE_RETENTION_POLICY_REMOVE:
+          remove_command_file = True
+          logger.info(
+            'Removing %s due to the command_file_retention_policy, %s',
+            json_path, command_file_retention_policy
+          )
+        elif command_file_retention_policy == self.config.COMMAND_FILE_RETENTION_POLICY_REMOVE_ON_SUCCESS:
+          if ret and ('exitcode' in ret):
+            exit_code = ret['exitcode']
+            if exit_code == 0:
+              remove_command_file = True
+              logger.info(
+                'Removing %s due to the command_file_retention_policy, %s, and exit code, %d',
+                json_path, command_file_retention_policy, exit_code
+              )
+            else:
+              remove_command_file = False
+              logger.info(
+                'Not removing %s due to the command_file_retention_policy, %s, and exit code, %d',
+                json_path, command_file_retention_policy, exit_code
+              )
+          else:
+            remove_command_file = False
+            logger.info(
+              'Not Removing %s due to the command_file_retention_policy, %s, and a missing exit code value',
+              json_path, command_file_retention_policy
+            )
+        else:
+          remove_command_file = False
```

Comment:
To follow the above pattern you may add logging here too about the command file is not being removed due to retention policy.

Reply:
Do we need to log this, though? I'd say only log the removals.

Reply:
I agree, I think just logging removals (which is different the today's behavior) is better... else the log will get really big during normal operations.  

Maybe the logging should be turned down for simple removals (is that possible in Python?) and generally logged for mis-fires - failure to remove due to error condition when the policy is remove_on_success.

Reply:
I'm convinced

Reply:
Should we really flood ambari-agent.log with this? Since this is a default behaviour.

Reply:
No, we should not; it was only a question and I got the answer for it...